src/Network/Network/AzureFirewall/ApplicationRule/NewAzureFirewallApplicationRuleCommand.cs

```diff
@@ -40,21 +36,59 @@ public class NewAzureFirewallApplicationRuleCommand : AzureFirewallBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Description { get; set; }
 
+        [Parameter(
+           Mandatory = true,
+           ParameterSetName = AzureFirewallApplicationRuleParameterSets.Default,
+           HelpMessage = "The source addresses of the rule")]
         [Parameter(
             Mandatory = false,
-            HelpMessage = "The source addresses of the rule")]
-        [ValidateNotNullOrEmpty]
+            ParameterSetName = AzureFirewallApplicationRuleParameterSets.TargetFqdn,
+            HelpMessage = "The target FQDNs of the rule")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = AzureFirewallApplicationRuleParameterSets.FqdnTag,
+            HelpMessage = "The FQDN Tags of the rule")]
```

Comment:
The help message is miss match with the parameter.

Reply:
Which line is this?

Reply:
Reordered parameter definition so the relevant name appears over the parameter